test/run_all.sh

```diff
@@ -4,76 +4,35 @@ set -e #-x
+LOG_PATH=$NGINX_PREFIX/logs
 
 
+# v24_features and v26_features fail now. They should be added
+# into this array with gh-144 fix.
+declare -a test_files=("basic_features" "v20_features" "v23_features"
+                       "v25_features" "v27_features")
 echo "[+] Basic test"
 for i in {1..10}; do
   echo "[+] try: $i"
-  $WORK_DIR/basic_features.py 1> /dev/null || (
-      echo "[-] $WORK_DIR/basic_features.py failed" && exit 1
-    )
-  $WORK_DIR/v20_features.py 1> /dev/null || (
-      echo "[-] $WORK_DIR/v20_features.py failed" && exit 1
-    )
-  $WORK_DIR/v23_features.py 1> /dev/null || (
-      echo "[-] $WORK_DIR/v23_features.py failed" && exit 1
-    )
-  # This test fails now. Should be returned with gh-144 fix.
-  # $WORK_DIR/v24_features.py 1> /dev/null || (
-  #     echo "[-] $WORK_DIR/v24_features.py failed" && exit 1
-  #   )
-#  $WORK_DIR/lua.py 1> /dev/null || (
-#      echo "[-] $WORK_DIR/lua.py failed" && exit 1
-#    )
-  $WORK_DIR/v25_features.py 1> /dev/null || (
-      echo "[-] $WORK_DIR/v25_features.py failed" && exit 1
-    )
-  # This test fails now. Should be returned with gh-144 fix.
-  # $WORK_DIR/v26_features.py 1> /dev/null || (
-  #     echo "[-] $WORK_DIR/v26_features.py failed" && exit 1
-  #   )
-  $WORK_DIR/v27_features.py 1> /dev/null || (
-      echo "[-] $WORK_DIR/v26_features.py failed" && exit 1
-    )
+  for test_file in "${test_files[@]}"; do
+    $WORK_DIR/$test_file.py 1>> $LOG_PATH/$test_file.log || (
+        echo "[-] $WORK_DIR/$test_file.py failed" && exit 1
+      )
+  done
 done
 echo '[+] OK'
 
 
 echo '[+] Parallel clients'
 clients_pids=
 for i in {1..3}; do
-  `$WORK_DIR/basic_features.py 1> /dev/null || (
-      echo "[-] $WORK_DIR/basic_features.py failed" && exit 1
-    )` &
-  clients_pids="$clients_pids $!"
-  `$WORK_DIR/v20_features.py 1> /dev/null || (
-      echo "[-] $WORK_DIR/v20_features.py failed" && exit 1
-    )` &
-  clients_pids="$clients_pids $!"
-  `$WORK_DIR/v23_features.py 1> /dev/null || (
-      echo "[-] $WORK_DIR/v23_features.py failed" && exit 1
-    )` &
-  # This test fails now. Should be returned with gh-144 fix.
-  # clients_pids="$clients_pids $!"
-  # `$WORK_DIR/v24_features.py 1> /dev/null || (
-  #     echo "[-] $WORK_DIR/v24_features.py failed" && exit 1
-  #   )` &
-#  clients_pids="$clients_pids $!"
-#  `$WORK_DIR/lua.py 1> /dev/null || (
-#      echo "[-] $WORK_DIR/lua.py failed" && exit 1
-#    )` &
-  $WORK_DIR/v25_features.py 1> /dev/null || (
-      echo "[-] $WORK_DIR/v25_features.py failed" && exit 1
-    ) &
-
-  # XXX It couldn't be work in parallel
-  #$WORK_DIR/v26_features.py 1> /dev/null || (
-  #    echo "[-] $WORK_DIR/v26_features.py failed" && exit 1
-  #  )
-  $WORK_DIR/v27_features.py 1> /dev/null || (
-      echo "[-] $WORK_DIR/v26_features.py failed" && exit 1
-    )
-  clients_pids="$clients_pids $!"
+  echo "[+] try: $i"
+  for test_file in "${test_files[@]}"; do
+    `$WORK_DIR/$test_file.py 1> $LOG_PATH/$test_file.log || (
```

Comment:
It'll rewrite logs from non-parallel run. I would name them differently.

BTW, you appending (`>>`) logs for non-parallel runs to existing logs, but replacing (`>`) old logs for parallel run. There should be some consistency.

Reply:
Done.

Reply:
I choose `>`.

Reply:
Parallel runs of one test still write to the same log file. It should contain an iteration number.

Reply:
Done.